test/datasets_utils.py

```diff
@@ -114,26 +114,61 @@ def inner_wrapper(*args, **kwargs):
     return outer_wrapper
 
 
-def test_all_configs(test):
+def test_all_configs(*, remove_duplicates=True, strict=False):
     """Decorator to run test against all configurations.
 
-    Add this as decorator to an arbitrary test to run it against all configurations. The current configuration is
-    provided as the first parameter:
+    Add this as decorator to an arbitrary test to run it against all configurations. This includes
+    :attr:`DatasetTestCase.DEFAULT_CONFIG` and :attr:`DatasetTestCase.ADDITIONAL_CONFIGS`.
+
+    The current configuration is provided as the first parameter for the test:
 
     .. code-block::
 
-        @test_all_configs
+        @test_all_configs()
         def test_foo(self, config):
             pass
+
+    .. note::
+
+        This will not preserve a potential ordering of the configuration or an inner ordering of a configuration.
+
+    Args:
+        remove_duplicates (bool): If ``True`` (default), tries to remove duplicate configurations. If any
+            :class:`Exception` is encountered, e.g. a configuration contains an unhashable value, the complete set of
+            configurations is used as fallback.
+        strict (bool): If ``True``, raise an :class:`Exception` during duplicate removing instead of using a fallback.
     """
 
-    @functools.wraps(test)
-    def wrapper(self):
-        for config in self.CONFIGS or (self._DEFAULT_CONFIG,):
-            with self.subTest(**config):
-                test(self, config)
+    def maybe_remove_duplicates(configs, *, strict):
+        try:
+            return [dict(config_) for config_ in set(tuple(sorted(config.items())) for config in configs)]
+        except Exception:
```

Comment:
Should we just expect a TypeError for unhashable parameters, instead of the more general Exception class, which I believe is discouraged?

Also nit: `raise from` tends to give better stack traces